homeassistant/components/android_ip_webcam/camera.py

```diff
@@ -2,43 +2,59 @@
 from __future__ import annotations
 
 from homeassistant.components.mjpeg import MjpegCamera, filter_urllib3_logging
-from homeassistant.const import HTTP_BASIC_AUTHENTICATION
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_NAME,
+    CONF_PASSWORD,
+    CONF_USERNAME,
+    HTTP_BASIC_AUTHENTICATION,
+)
 from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import DeviceInfo
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
-from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
 
+from .const import DOMAIN
+from .coordinator import AndroidIPCamDataUpdateCoordinator
 
-async def async_setup_platform(
+
+async def async_setup_entry(
     hass: HomeAssistant,
-    config: ConfigType,
+    config_entry: ConfigEntry,
     async_add_entities: AddEntitiesCallback,
-    discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
-    """Set up the IP Webcam camera."""
-    if discovery_info is None:
-        return
-
+    """Set up the IP Webcam camera from config entry."""
     filter_urllib3_logging()
-    async_add_entities([IPWebcamCamera(**discovery_info)])
+    coordinator: AndroidIPCamDataUpdateCoordinator = hass.data[DOMAIN][
+        config_entry.entry_id
+    ]
+
+    async_add_entities([IPWebcamCamera(coordinator)])
 
 
 class IPWebcamCamera(MjpegCamera):
     """Representation of a IP Webcam camera."""
 
-    def __init__(
-        self,
-        name: str,
-        mjpeg_url: str,
-        still_image_url: str,
-        username: str | None = None,
-        password: str = "",
-    ) -> None:
+    _attr_has_entity_name = True
+
+    def __init__(self, coordinator: AndroidIPCamDataUpdateCoordinator) -> None:
         """Initialize the camera."""
+        # keep imported name until YAML is removed
+        name = (
+            coordinator.config_entry.data.get(CONF_NAME)
+            or coordinator.config_entry.data[CONF_HOST]
+        )
+
         super().__init__(
             name=name,
```

Comment:
If we set `_attr_has_entity_name` to `True` we should not set an entity name for the camera entity as it's the main entity of this device. It should just get the device name. So the entity name should be `None` in that case.

For imported entities we shouldn't set `_attr_has_entity_name` to `True` to avoid a breaking change as there's no unique id already in place.